src/libutil/args.cc

```diff
@@ -69,6 +76,36 @@ void Args::parseCmdline(const Strings & _cmdline)
     }
 
     bool argsSeen = false;
+
+    // Heuristic to see if we're invoked as a shebang script, namely,
```

Comment:
Since this is very similar to the code in nix-build.cc, it would be nice to factor out the commonality.